src/Maestro/Maestro.Common/AppCredential.cs

```diff
@@ -0,0 +1,156 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Core;
+using Azure.Identity;
+
+namespace Microsoft.DotNet.Maestro.Common;
+
+/// <summary>
+/// A credential for authenticating against Azure applications.
+/// </summary>
+public class AppCredential : TokenCredential
+{
+    private static readonly string AUTH_CACHE = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".darc");
```

Comment:
Didn't this use to be public so that we can customize it?

Reply:
Yeah, it was. Thanks for noticing, will change